wp/lib/bap_wp/src/constraint.ml

```diff
@@ -49,6 +50,13 @@ let get_model_exn (solver : Solver.solver) : Model.model =
 let goal_to_string (g : goal) : string =
   Format.sprintf "%s: %s%!" g.goal_name (Expr.to_string (Expr.simplify g.goal_val None))
 
+let expr_to_hex (exp : z3_expr) : string =
+  let decimal = exp |> BV.numeral_to_string |> Int.of_string in
```

Comment:
Uh oh. Can this ever overflow? Isn't this 63 bits?

Reply:
Uh oh. It might be able to overflow.